src/SDKs/Azure.ApplicationModel.Configuration/data-plane/Azure.ApplicationModel.Configuration.Performance/ConfigurationClientBenchmark.cs

```diff
@@ -0,0 +1,66 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Net;
+using System.Net.Http;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Azure.Core.Pipeline;
+using BenchmarkDotNet.Attributes;
+
+namespace Azure.ApplicationModel.Configuration.Performance
+{
+    [MemoryDiagnoser]
+    public class ConfigurationClientBenchmark
+    {
+        private static readonly string ConnectionString = "Endpoint=https://contoso.appconfig.io;Id=b1d9b31;Secret=aabbccdd";
+
+        private static readonly MockClientHandler GetResponseMessage = new MockClientHandler(request => new HttpResponseMessage((HttpStatusCode)200)
+        {
+            Content = new ByteArrayContent(_getBytes)
+        });
+
+        private static readonly ConfigurationClient ConfigurationClient = new ConfigurationClient(ConnectionString, new ConfigurationClientOptions()
+        {
+            Transport = new HttpClientTransport(new HttpClient(GetResponseMessage))
+        });
+
+        private static byte[] _getBytes = Encoding.UTF8.GetBytes(@"{
+  ""etag"": ""4f6dd610dd5e4deebc7fbaef685fb903"",
+  ""key"": ""key"",
+  ""label"": ""label"",
+  ""content_type"": null,
+  ""value"": ""example value"",
+  ""last_modified"": ""2017-12-05T02:41:26+00:00"",
+  ""locked"": false,
+  ""tags"": {
+    ""t1"": ""value1"",
+    ""t2"": ""value2""
+  }
+}");
+
+        [Benchmark]
+        public async Task GetSecretBenchmarkCore()
```

Comment:
>public async Task GetSecretBenchmarkCore() [](start = 8, length = 42)

nit:  why is this called GetSecretBenchmarkCore?  Is this because that was the name of a previously existing benchmark you were comparing it to?  I saw GetSecretBenchmarkAz in the comments on this PR, but I couldn't find the implementation of this.  Not a big deal, but at first I thought this was comparing to the performance of Microsoft.Extensions.Configuration.AzureKeyVault.

Reply:
I'll fix the name. I didn't include the `GetSecretBenchmarkAz` in this PR because it relies on a locally built version of `Microsoft.Azure.AppConfiguration.AzconfigClient` as all of it's APIs are internal and the package is not published to any publically accessible feeds.